internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -1326,6 +1328,193 @@ func TestMachineSetReconciler_syncMachines(t *testing.T) {
 	}, 5*time.Second).Should(Succeed())
 }
 
+func TestMachineSetReconciler_reconcileUnhealthyMachines(t *testing.T) {
+	t.Run("should delete unhealthy machines if preflight checks pass", func(t *testing.T) {
+		defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.MachineSetPreflightChecks, true)()
+
+		g := NewWithT(t)
+
+		controlPlaneStable := builder.ControlPlane("default", "cp1").
+			WithVersion("v1.26.2").
+			WithStatusFields(map[string]interface{}{
+				"status.version": "v1.26.2",
+			}).
+			Build()
+		cluster := &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster",
+				Namespace: "default",
+			},
+			Spec: clusterv1.ClusterSpec{
+				ControlPlaneRef: contract.ObjToRef(controlPlaneStable),
+			},
+		}
+		machineSet := &clusterv1.MachineSet{}
+
+		unhealthyMachine := &clusterv1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "unhealthy-machine",
+				Namespace: "default",
+			},
+			Status: clusterv1.MachineStatus{
+				Conditions: []clusterv1.Condition{
+					{
+						Type:   clusterv1.MachineOwnerRemediatedCondition,
+						Status: corev1.ConditionFalse,
+					},
+				},
+			},
+		}
+		healthyMachine := &clusterv1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "healthy-machine",
+				Namespace: "default",
+			},
+		}
+
+		machines := []*clusterv1.Machine{unhealthyMachine, healthyMachine}
+
+		fakeClient := fake.NewClientBuilder().WithObjects(controlPlaneStable, unhealthyMachine, healthyMachine).Build()
+		r := &Reconciler{Client: fakeClient}
+		_, err := r.reconcileUnhealthyMachines(ctx, cluster, machineSet, machines)
+		g.Expect(err).To(BeNil())
+		// Verify the unhealthy machine is deleted.
+		g.Eventually(func(g Gomega) {
```

Comment:
There should be no need to use eventually and consistently with the fake client, right? (same below)

Reply:
You are correct. With fake client the results should be instant. We don't need the eventually/consistently.